doc/development.md

```diff
@@ -89,4 +89,205 @@
 architectures.
 
 When you push patches to GitHub, CI will run its tests according to the
-configuration in the `.github/workflows/ci.yml` file.
+configuration in the `.github/workflows/ci.yml` file.
+
+## Coding guidelines
+
+The coding conventions in Vdsm are generally aligned with the
+[PEP 8](https://peps.python.org/pep-0008/) standard guidelines.
+Code is linted with `pylint` based in the `/.pylintrc` file.
+You can lint the code before submitting a patch by doing:
+
+    make lint
+
+However, some code conventions are not explicitly enforced by the linter due
+to legacy code and inconsistent styles. But new code shall adhere to those
+rules, and old code, should be updated as possible when patched.
+
+In this section, we will cover some of the most important conventions and
+preferences for contributing to Vdsm.
+
+### Imports
+
+First import standard library modules, then 3rd party modules, and finally
+project modules.
+Separate standard library modules and 3rd party modules with a blank line.
+Keep alphabetical order within each block.
+
+Less specific imports:
+
+    from vdsm.foo import baz
+
+Should come before more specific imports like:
+
+    from vdsm.foo.bar import baz
+
+Accepted, but better to avoid:
+- Relative name imports
+- Import many modules in a comma separated list
+- Alias an imported module with non-standard abbreviation
+
+### Indentation
+
+Use 4 spaces per indentation level.
+
+Continuation lines should align wrapped elements either vertically using
+Python's implicit line joining inside parentheses, brackets and braces, or
+using a hanging indent. When using a hanging indent, the following should
+be considered:
+- There should be no arguments on the first line.
+- Further indentation should be used to clearly distinguish itself as a continuation line.
+- Hanging indentation should stick to the 4-space rule.
+```python
+# Correct
+# Add 4 spaces (an extra level of indentation) to distinguish arguments from the rest.
+foo = long_function_name(
+        var_one, var_two, var_three, var_four)
+
+# Also with if-statements.
+if (condition_1 and
+        condition_2):
```

Comment:
This style of condition misalignment by definition is horrible but it seems to be quite popular and I've experienced insistence on it in Vdsm reviews.  I have better things to do than arguing about broken indentation so I can live with this rule.

Reply:
Nothing here is written in stone, if there is a place for arguing is now :) The target is that we all reach an agreement and are comfortable with the result. I know, ambitious goal :P

That said, the reasoning for the extra 4 spaces here is to distinguish it from the code inside the if-statement. Text from PEP:
```
(...) the combination of a two character keyword (i.e. if), plus a single space, plus an opening parenthesis
creates a natural 4-space indent for the subsequent lines of the multiline conditional.
This can produce a visual conflict with the indented suite of code nested inside the if-statement,
which would also naturally be indented to 4 spaces.
```
However, PEP does not enforce it, and thus we can also not enforce it if we don't like it.

Reply:
There is no problem to work around :-)

Reply:
I think I will relax it to a recommendation only, no need to enforce it really.

To work around it, I think would favor evaluating the conditions separately:
```python
condition_name = condition_1 and condition_2 [and ...]
if condition_name:
    do_something()
```
Which can be split into multiple statements, or multiple lines, easily.